### PR TITLE
Blacklist unsafe redirect URLs

### PR DESCRIPTION
**Pull request by dvoytenko**
b/122875496

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -23,6 +23,11 @@ const ABORT_ERR_NAME = 'AbortError';
 /** DOMException.ABORT_ERR = 20 */
 const ABORT_ERR_CODE = 20;
 
+/** Normal protocol shape is "xyz:" */
+const PROTOCOL_RE = /.*\:$/;
+
+const SCHEME_REDIRECT_BLACKLIST_RE = /javascript|script|data/i;
```
Comment (cramforce) [inline src/utils.js]: Why can't this be an allow-list instead of a disallow-list?

File: src/utils.js [hunk 2]
```diff
@@ -23,6 +23,11 @@ const ABORT_ERR_NAME = 'AbortError';
 /** DOMException.ABORT_ERR = 20 */
 const ABORT_ERR_CODE = 20;
 
+/** Normal protocol shape is "xyz:" */
+const PROTOCOL_RE = /.*\:$/;
```
Comment (cramforce) [inline src/utils.js]: When would this be needed?

File: src/utils.js (hunk 1 above)
Comment (dvoytenko) [reply src/utils.js]: Not super comfortable with this either. But I wanted to enable custom protocols (in- and out-of-native apps) by default. There's a separate `accept` API that requires more conformance before the actual data is sent over. Thus, in this form, this API is mainly for cancelations and errors.

**Comment by cramforce** [comment]
Makes sense. Please document as this is definitely a bad practice.

**Comment by kaipakartik** [comment]
Is returnUrl the only place where this can happen?

Are all the other parameters safe?

**Comment by kaipakartik** [comment]
Also is there a better way to do this? Maybe using safe html or something like that instead?

**Comment by dvoytenko** [comment]
@cramforce @kaipakartik actually, i have a better idea. should have it up shortly in a separate pr.

**Pull request closed without merging**